src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1055,6 +1081,23 @@ qreal TorrentHandle::maxRatio(bool *usesGlobalRatio) const
     return ratioLimit;
 }
 
+int TorrentHandle::maxSeedingTime(bool *usesGlobalSeedingTime) const
```

Comment:
I tried to follow the same coding style that was already there.  Look for maxRatio() and other similar functions that has the same implementation.  Remember this PR was already approved, but never merged.  This is just a rebase actually, but I really appreciate your comments, they have been very helpful.